xmake/rules/c++/modules/modules_support/msvc.lua

```diff
@@ -399,9 +449,23 @@ function get_scandependenciesflag(target)
                 end
                 return ok, errors
             end}) then
-            scandependenciesflag = "/scanDependencies"
+            scandependenciesflag = "-scanDependencies"
         end
         _g.scandependenciesflag = scandependenciesflag or false
     end
     return scandependenciesflag or nil
 end
+
+-- append .obj to final link
+function append_headerunits_objectfiles(target)
+    local linker_file = _get_linker_argument_file()
```

Comment:
Why use file? Can't we just cache them in memory using target:data_set() or memcache()?

Reply:
other targets may need them

Reply:
> other targets may need them so for now, i add all the header unit objectfiles to all targets

you can use `common.memcache()`

Reply:
memcache is clear between two invocation of xmake b ?

Reply:
i need to use globalcache i think

Reply:
why? Will all projects share it?

Reply:
>     local linker_file = path.join(config.buildir(), "linkargs.txt")

I see that you are only using the local cache

Reply:
why localcache doesn't store anything ?

```lua
    local cache = common.localcache():get("module_mapper") or {}
    for _, flag in ipairs(cache) do
        if flag:startswith(argument .. " " .. module) then
            return
        end
    end
    table.insert(cache, format("%s %s", argument, module))
    common.localcache():set("module_mapper", cache)
    common.localcache():save("module_mapper")
```

Reply:
.xmake/macosx/x86_64/cache/cxxmodules

Reply:
nvm i was using memcache instead of the localcache in build_modules =P